backend_jobs/ingestion_pipeline/providers/image_provider_flickr.py

```diff
@@ -17,114 +17,85 @@
+            extras = 'url_c, geo, date_upload, date_taken, original_format, \
+                owner_name, original_format',
+            per_page = _NUM_OF_IMAGES,
+            page = page_number,
+            sort = 'relevance')
+        return photos[0] # return Element 'photos'
 
     def get_num_of_pages(self):
         photos = self.get_images(1)
         return int(photos.attrib['pages'])
 
     def get_image_attributes(self, element):
-        image_arrributes=ImageAttributes(
+        image_attributes=ImageAttributes(
             url = element.get('url_c'),
-            image_id = self._generate_image_id_with_prefix(element.get('id')),
+            image_id = None,
             image_type = ImageType.CAMERA,
-            date_shot = get_date(element),
-            coordinates = get_coordinates(element),
+            date_shot = _get_date(element),
             format = element.get('originalformat'),
             attribution = element.get('ownername'),
-            resolution = get_resolution(element))
-        return image_arrributes
-
-    def get_url_for_max_resolution(self, resolution, image):
-        # This function modifies the url to get the wonted resolution,
+            latitude = float(element.get('latitude')),
+            longitude = float(element.get('longitude')),
+            #Extracts width from element, if the width equals str converts to int.
+            width_pixels = int(element.get('width_c')) if element.get('width_c') else None,
+            #Extracts height from element, if the height equals str converts to int.
+            height_pixels = int(element.get('height_c')) if element.get('height_c') else None)
+        return image_attributes
+
+    def get_url_for_min_resolution(self,  min_height, min_width, image):
+        # This function modifies the url to get the requested resolution,
         # in flickr the resolution is represented in the url.
         # See details at https://www.flickr.com/services/api/misc.urls.html
         url = image['url']
-        max_resolution =  max(resolution['height'],resolution['width'])
-        # Maping between the char that represents max resultion and the max resultion.
-        flickr_resolution_map = {75:'s',100:'t',150	:'q',240:'m',320:'n',400:'w',
-        500:'',640:'z',800:'c',1024:'b',1600:'n',2048:'k'}
+        max_resolution =  max(min_height, min_width)
+        # Maping between the char that represents max resolution and the max resolution.
+        flickr_resolution_map = {2048: 'k', 1600: 'n', 1024: 'b', 800: 'c', 640: 'z',500: '',
+        400: 'w', 320: 'n', 240: 'm', 150: 'q', 100: 't', 75: 's'}
         for key in flickr_resolution_map:
-            if  max_resolution <= key:
+            if  max_resolution >= key:
                 if key == 500:
-                    # Removing the char that represents the resultion to get max resultion of 500.
+                    # Removing the char that represents the resolution to get max resolution of 500.
```

Comment:
picking up my other comment... i don't think this code is correct. we don't know what the image.url looks like - how do we know whether it has a resolution marker in its url or not?

Reply:
seems correct at high level, but will need extensive test suite to try every corner case